pyrogram/client/client.py

```diff
@@ -527,53 +530,56 @@ def default_phone_number_callback():
 
             self.phone_number = self.phone_number.strip("+")
 
-            try:
-                r = self.send(
-                    functions.auth.SendCode(
-                        self.phone_number,
-                        self.api_id,
-                        self.api_hash
+            if phone_code is None or phone_code_hash is None:
+                try:
+                    r = self.send(
+                        functions.auth.SendCode(
+                            self.phone_number,
+                            self.api_id,
+                            self.api_hash
+                        )
                     )
```

Comment:
There are some huge changes because I have wrapped phone code hash getting block in the if statement